File: mambo-protocol/src/main/scala/org/mambo/protocol/types/game/context/roleplay/job/JobDescription.scala





















// Generated on 05/03/2013 10:35:05
// Scala version by Blackrush
package org.mambo.protocol.types

import org.mambo.core.io._
import org.mambo.protocol.{Type => BaseType, _}
import org.mambo.protocol.enums._

trait JobDescription extends TypeDeserializer {
  val typeId = 101

  protected trait Type extends BaseType with Serializable {
    val typeId = JobDescription.typeId

    val jobId: Byte
    val skills: Seq[SkillActionDescription.TypeImpl]

    def serialize(buf: Buffer) {
      buf.writeByte(jobId)
      buf.writeUShort(skills.length)
      for (entry <- skills) {
        buf.writeUShort(entry.typeId)
        entry.serialize(buf)
      }
    }

  }

  protected def create(jobId: Byte, skills: Seq[SkillActionDescription.TypeImpl]): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val jobId = buf.readByte()
    val skills = {
      val builder = collection.immutable.Seq.newBuilder[SkillActionDescription.TypeImpl]
      for (i <- (0 to buf.readUShort())) {
        builder += types.get[SkillActionDescription.TypeImpl](buf.readShort());
      }
      builder.result()
    }

    create(jobId, skills)
  }

}

object JobDescription extends JobDescription {
  case class TypeImpl(jobId: Byte, skills: Seq[SkillActionDescription.TypeImpl]) extends Type
  override def create(jobId: Byte, skills: Seq[SkillActionDescription.TypeImpl]) = TypeImpl(jobId, skills)

  def unapply(msg: TypeImpl) = TypeImpl.unapply(msg)
  def apply(jobId: Byte, skills: Seq[SkillActionDescription.TypeImpl]) = create(jobId, skills)

  type Target = TypeImpl

}

File: mambo-protocol/src/main/java/org/mambo/protocol/types/JobDescription.java


// Generated on 05/08/2013 19:38:01
package org.mambo.protocol.types;

import java.util.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class JobDescription extends NetworkType {
    public static final short TYPE_ID = 101;
    
    @Override
    public short getTypeId() {
        return TYPE_ID;
    }
    
    public byte jobId;
    public SkillActionDescription[] skills;
    
    public JobDescription() { }
    
    public JobDescription(byte jobId, SkillActionDescription[] skills) {
        this.jobId = jobId;
        this.skills = skills;
    }
    
    @Override
    public void serialize(Buffer buf) {
        buf.writeByte(jobId);
        buf.writeUShort(skills.length);
        for (SkillActionDescription entry : skills) {
            buf.writeShort(entry.getTypeId());
            entry.serialize(buf);
        }
    }
    
    @Override
    public void deserialize(Buffer buf) {
        jobId = buf.readByte();
        if (jobId < 0)
            throw new RuntimeException("Forbidden value on jobId = " + jobId + ", it doesn't respect the following condition : jobId < 0");
        int limit = buf.readUShort();
        skills = new SkillActionDescription[limit];
        for (int i = 0; i < limit; i++) {
            skills[i] = ProtocolTypeManager.getInstance().build(buf.readShort());
            skills[i].deserialize(buf);
        }
    }
    
}
